### PR TITLE
Show error if add-on is using Mozilla trademarks incorrectly.

### LINKED ISSUE #6693: Show error if add-on is using Mozilla trademarks incorrectly
This is something the legacy validator does, but apparently is not applied consistently for all add-ons, or for all cases. The rule is that you can only use `Firefox` or `Mozilla` in an add-on name if they are used at the end, in the form `for Firefox`, or `for Mozilla`, _or_ if the submitter has an account associated with an `@mozilla.com` or `@mozilla.org` email address.

We need this to be applied to WebExtension add-ons, and we also need to ensure this rule is applied during the submission flow and the developer hub, where the name can be changed by the developer. This is particularly important now that we have auto-approval turned on.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #6693

Errors for misusing the trademark:

screenshot from 2018-01-23 18-03-19

screenshot from 2018-01-23 18-20-32

Perfectly fine using "for Mozilla" postfix:

screenshot from 2018-01-23 18-20-54

### CODE REVIEW

File: src/olympia/addons/forms.py [hunk 1]
```diff
@@ -150,9 +161,6 @@ def __init__(self, *args, **kw):
             self.fields['tags'].initial = ', '.join(
                 self.get_tags(self.instance))
 
-    def clean_slug(self):
-        return clean_addon_slug(self.cleaned_data['slug'], self.instance)
-
```
Comment (EnTeQuAk) [inline src/olympia/addons/forms.py]: This is duplicated and already defined in the base class

File: src/olympia/addons/utils.py [hunk 2]
```diff
@@ -92,3 +96,43 @@ def get_creatured_ids(category, lang=None):
...
+            contains_trademark_symbol and not
+            name.endswith(tuple(
+                'for {}'.format(symbol)
+                for symbol in amo.MOZILLA_TRADEMARK_SYMBOLS)))
+
+        if violates_trademark:
+            raise forms.ValidationError(ugettext(
+                u'Add-on names cannot contain the Mozilla or '
+                u'Firefox trademarks. These names should not be '
+                u'contained in add-on names if at all possible.'))
```
Comment (jrbenny35) [inline src/olympia/addons/utils.py]: I feel like this is redundant. The first line ```Add-on names cannot contain the Mozilla or Firefox trademarks.``` is fine. And maybe split up 'cannot' into 'can not'.

Comment (EnTeQuAk) [reply src/olympia/addons/utils.py]: I'll just cite http://www.write.com/writing-guides/general-writing/word-choice/cannot-versus-can-not/ on the "cannot" thing and yeah, I think I agree about the redundancy. I'll remove it

Comment (jrbenny35) [reply src/olympia/addons/utils.py]: That is a great resource!

**Pull request merged**